lib/services/local/access_list.go

```diff
@@ -302,6 +321,154 @@ func (a *AccessListService) AccessRequestPromote(_ context.Context, _ *accesslis
+	err = a.service.RunWhileLocked(ctx, lockName(accessList), accessListLockTTL, func(ctx context.Context, _ backend.Backend) error {
+		_, err := a.service.GetResource(ctx, accessList)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		reviews, nextToken, err = a.reviewService.WithPrefix(accessList).ListResources(ctx, pageSize, pageToken)
+		return trace.Wrap(err)
+	})
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	return reviews, nextToken, nil
+}
+
+// CreateAccessListReview will create a new review for an access list.
+func (a *AccessListService) CreateAccessListReview(ctx context.Context, review *accesslist.Review) (createdReview *accesslist.Review, err error) {
+	reviewName := strings.ToLower(uuid.New().String())
+	createdReview, err = accesslist.NewReview(header.Metadata{
+		Name: reviewName,
+	}, accesslist.ReviewSpec{
+		AccessList: review.Spec.AccessList,
+		Reviewers:  review.Spec.Reviewers,
+		ReviewDate: review.Spec.ReviewDate,
+		Changes:    review.Spec.Changes,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = a.service.RunWhileLocked(ctx, lockName(review.Spec.AccessList), accessListLockTTL, func(ctx context.Context, _ backend.Backend) error {
+		accessList, err := a.service.GetResource(ctx, review.Spec.AccessList)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		if createdReview.Spec.Changes.MembershipRequirementsChanged != nil {
+			if accessListRequiresEqual(*createdReview.Spec.Changes.MembershipRequirementsChanged, accessList.Spec.MembershipRequires) {
+				createdReview.Spec.Changes.MembershipRequirementsChanged = nil
+			} else {
+				accessList.Spec.MembershipRequires = *review.Spec.Changes.MembershipRequirementsChanged
+			}
+		}
+
+		if createdReview.Spec.Changes.ReviewFrequencyChanged != 0 {
+			if createdReview.Spec.Changes.ReviewFrequencyChanged == accessList.Spec.Audit.Recurrence.Frequency {
+				createdReview.Spec.Changes.ReviewFrequencyChanged = 0
+			} else {
+				accessList.Spec.Audit.Recurrence.Frequency = review.Spec.Changes.ReviewFrequencyChanged
+			}
+		}
+
+		if createdReview.Spec.Changes.ReviewDayOfMonthChanged != 0 {
+			if createdReview.Spec.Changes.ReviewDayOfMonthChanged == accessList.Spec.Audit.Recurrence.DayOfMonth {
+				createdReview.Spec.Changes.ReviewDayOfMonthChanged = 0
+			} else {
+				accessList.Spec.Audit.Recurrence.DayOfMonth = review.Spec.Changes.ReviewDayOfMonthChanged
+			}
+		}
+
+		review.SetName(reviewName)
+		if err := a.reviewService.WithPrefix(review.Spec.AccessList).CreateResource(ctx, createdReview); err != nil {
+			return trace.Wrap(err)
+		}
+
+		accessList.Spec.Audit.NextAuditDate = services.SelectNextReviewDate(accessList)
+
+		for _, removedMember := range review.Spec.Changes.RemovedMembers {
+			if err := a.memberService.WithPrefix(createdReview.Spec.AccessList).DeleteResource(ctx, removedMember); err != nil {
```

Comment:
```suggestion
			if err := a.memberService.WithPrefix(review.Spec.AccessList).DeleteResource(ctx, removedMember); err != nil {
```
nit: I know it's the exact same thing, but a couple lines above we used review instead of createdReview and I had to think about it